fieldC/dynaField.c

```diff
@@ -0,0 +1,211 @@
+double freqAtt, attF0, att;
+int numCYC = 50;
+int numSteps;
+double *intensity;
+point_type *points;
+
+/*
+for (i = 0; i < 13; i++)
+        fprintf(stderr, "in dynaField, node %d is %d, %f, %f, %f\n", i, params.pointsAndNodes[i].nodeID, params.pointsAndNodes[i].x, params.pointsAndNodes[i].y, params.pointsAndNodes[i].z);
+*/
+
+/* how do I do check_add_probes? */
+
+/*
+ * initialize Field II; field_init is in the provided c library; the '-1'
+ * arg supresses ascii output.
+ */
+	
+	sys_con = field_init(-1);
+
+/* set transducer-independent parameters */
+
+	set_field("c", params.soundSpeed);
+	set_field("fs", params.samplingFrequency);
+	set_field("threads", params.threads);
+	fprintf(stderr, "PARALLEL THREADS: %d\n", threads);
+
+/* get info from JSON */
+	input = fopen("./c52Fixed.json","rb");
+	fseek(input,0,SEEK_END);
+	len=ftell(input );
+	fseek(input,0,SEEK_SET);
+	data=(char*)malloc(len+1);
+	fread(data,1,len,input);
+	fclose(input);
+
+	probeInfo = cJSON_Parse(data);
+	if (!probeInfo) {
+		printf("Error before: [%s]\n",cJSON_GetErrorPtr());
+		}
+
+	else {
+		out=cJSON_Print(probeInfo);
+/* 		printf("%s\n",out); */
+		free(out);
+		}
+
+	fprintf(stderr, "array size %d\n", cJSON_GetArraySize(probeInfo));
+
+	no_elements = cJSON_GetObjectItem(probeInfo, "no_elements")->valueint;
+	no_sub_x = cJSON_GetObjectItem(probeInfo, "no_sub_x")->valueint;
+	no_sub_y = cJSON_GetObjectItem(probeInfo, "no_sub_y")->valueint;
+
+	fprintf(stderr, "elements %d subX %d subY %d\n", no_elements, no_sub_x,
+		no_sub_y);
+
+	width = cJSON_GetObjectItem(probeInfo, "width")->valuedouble;
+	fprintf(stderr, "width %f\n", width);
+	height = cJSON_GetObjectItem(probeInfo, "height")->valuedouble;
+	kerf = cJSON_GetObjectItem(probeInfo, "kerf")->valuedouble;
+	Rconvex = cJSON_GetObjectItem(probeInfo, "Rconvex")->valuedouble;
+	Rfocus = cJSON_GetObjectItem(probeInfo, "Rfocus")->valuedouble;
+
+/*
+	fprintf(stderr, "type %s\n", cJSON_GetObjectItem(probeInfo, "probe_type")->valuestring);
+	commands = cJSON_GetObjectItem(probeInfo, "commands");
+	fprintf(stderr, "Th %s\n", cJSON_GetObjectItem(commands, "Th")->valuestring);
+	fprintf(stderr, "impulse response command %s\n", cJSON_GetObjectItem(commands, "impulseResponse")->valuestring);
+	impulseResponse = cJSON_GetObjectItem(probeInfo, "impulse_response");
+	fprintf(stderr, "center freq %d\n", cJSON_GetObjectItem(impulseResponse, "f0")->valueint);
+	fprintf(stderr, "phase %d\n", cJSON_GetObjectItem(impulseResponse, "phase")->valueint);
+	fprintf(stderr, "bw %d\n", cJSON_GetObjectItem(impulseResponse, "bw")->valueint);
+*/
+
+/* I think the next thing is to set Th, impulse */
+
+	Th = xdc_convex_focused_array(no_elements,width,height,kerf,Rconvex,Rfocus,no_sub_x,no_sub_y,params.focus);
+
+	fprintf(stderr, "#elements %d\n", Th->no_elements);
+	fprintf(stderr, "num apertures from sys_con %d\n", sys_con->No_apertures);
+
+	info = "rect";
+
+/* 	foo = (double *)malloc(26*1344*sizeof(double)); */
+	
+	xdc_get(Th, info, foo);
```

Comment:
@ndanieley says something here isn't working as expected (or we don't know what it expected)

Reply:
LGTM!